src/main/java/com/example/demo/service/PatientService.java

```diff
@@ -0,0 +1,54 @@
+package com.example.demo.service;
+
+import com.example.demo.model.Patient;
+import com.example.demo.repository.PatientRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class PatientService {
+    PatientRepository patientRepository;
+
+    public PatientService(PatientRepository patientRepository) {
+        this.patientRepository = patientRepository;
+    }
+
+    public List<Patient> getAllPatients() {
+        return patientRepository.getPatients();
+    }
+
+    public Patient getPatientByEmail(String email) {
+        List<Patient> allPatients = getAllPatients();
+
+        return allPatients.stream()
+                .filter(patient -> patient.getEmail().equalsIgnoreCase(email))
+                .findAny()
+                .get();
+    }
+
+    public Patient addNewPatient(Patient patient) {
+        return patientRepository.save(patient);
+    }
+
+    public void removePatientByEmail(String email) {
+        Patient patientByEmail = getPatientByEmail(email);
```

Comment:
Optional<Patient> patientByEmail = patienRepository.getPatientByEmail(email).orElseThrow(() -> new PatientNotFoundException("Patient with given email does not exist."))